docs: add IDKit invite-code mode page (#111)
worldcoin/developer-docs: world-id/idkit/verification-flows.mdx

---
title: "Verification flows"
description: "Understand the three verification paths a user can take when your app requests a World ID proof — hot, warm, and cold."
"og:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
"twitter:image": "https://raw.githubusercontent.com/worldcoin/developer-docs/main/images/docs/docs-meta.png"
---

When your app requests a World ID proof, the user is taken through one of three flows based on two factors: whether they have World App installed, and whether they already hold the credential you're requesting.

| Flow | World App installed | Has credential | What happens |
| --- | --- | --- | --- |
| Hot | Yes | Yes | World App opens, displays a proof consent, and the user approves. Typically under 10 seconds. |
| Warm | Yes | No | World App opens and walks the user through credential enrollment (e.g., PoH, Face, or NFC). Once issued, a proof consent appears and the user approves. |
| Cold | No | No | The user must install World App first. The experience differs by platform — see below. |

## Cold flow

> The cold flow requires the most steps and works differently on each platform.

Author: Introduce the idea of deferred deeplinking and how the cold flow has problems due to deferred deeplinking on the various platforms.

### Android

On Android, deep linking preserves context through the install process. After the user downloads World App from the Play Store and opens it, the original verification context is carried forward — World App picks up where the user left off and routes them directly into the credential enrollment flow. No re-trigger from your app is needed.

### iOS

iOS does not preserve context through the App Store install. This makes the cold flow higher friction and more prone to drop-off.

#### Default behavior

> 1. The user downloads World App from the App Store and completes account creation onboarding.

Author: ```suggestion
1. The user downloads World App from the App Store and completes account creatio.
```

2. Your app triggers an IDKit verification request.

> 3. World App opens and takes the user through a hot or warm flow.

Author: Add the step for how the user picks up the request post-install manually

4. The proof consent appears, the user approves, and the proof is returned to your app.

Because iOS does not preserve context through the App Store install, your app should direct the user to install World App before issuing the IDKit request.

#### With the App Clip (document enrollment)

When your app requests a document credential (e.g., NFC passport or national ID), the World App [App Clip](https://developer.apple.com/app-clips/) lets the user begin enrollment before the full app is installed:

1. Your app triggers an IDKit verification request for a document credential.
2. The user is presented with the World App App Clip.
3. The App Clip walks the user through country and document selection (passport, national ID, or My Number Card).

> 4. The user scans their document via NFC inside the App Clip.

Author: That's not correct.

> 5. The App Clip persists the scan data and selected document to a shared container, then prompts the user to install World App.

Author: This is half-correct.

6. The user installs World App from the App Store.
7. World App reads the shared container, completes account creation onboarding, and finishes credential enrollment using the data collected by the App Clip.

> 8. The proof consent appears, the user approves, and the proof is returned to your app.

Author: Let's add some diagrams

Because the App Clip and World App share a data container, the document scan and selection survive the install — the user does not need to re-scan or re-select their document after installing the full app.

#### With invite-code mode

Invite-code mode displays a short 6-character code in your app that the user enters into World App. World App treats the code as an entry point to the in-app onboarding flows the user needs to complete in order to satisfy your IDKit request, then returns the proof.

<Note>

>   Invite-code mode is only supported by World App on iOS.

Author: ```suggestion
  Invite-code mode is only supported by World App only applies to iOS as Android preserves deferred deeplinking context via the Google Play store.
```

</Note>

<Note>
  Only the `selfieCheckLegacy` preset is supported today.
</Note>

1. Your app triggers an IDKit invite-code request.

> 2. Your app opens the URL that IDKit provides. The `world.org/verify` landing page displays the invite code and a QR code for the request.

Author: ```suggestion
2. Your app opens the URL that IDKit provides. A landing page displays the invite code and a QR code for the request.
```

3. The user downloads World App from the App Store.
4. The user opens World App and completes account creation onboarding.
5. The user enters the invite code.
6. World App resolves the code, picks up the original verification context, and walks the user through credential enrollment.
7. The proof consent appears, the user approves, and the proof is returned to your app.

The 6-character code persists across the App Store install, so once World App is installed and onboarded the user can resume the verification flow without returning to your app first. This also covers cross-device scenarios (e.g., a desktop browser displaying the QR for the user's phone) where deep linking cannot carry context.

##### Code format

- 6 characters, case-insensitive.

> - Separators are ignored on entry. Display as `ABC-DEF` if you like; the canonical form has no separator.

Author: ```suggestion

```

##### Lifecycle

> - Codes expire after a short TTL (currently ten minutes). Pair the code with a countdown driven by `expiresAt`.

Author: ```suggestion
- Codes expire after a short TTL (currently ten minutes).
```
